docs/DeveloperGuide.md

```diff
@@ -310,20 +310,70 @@ The following activity diagram summarizes what happens when a user executes a ne
 
 _{more aspects and alternatives to be added}_
 
-### \[Proposed\] Reading an individual event feature
+### Reading an individual event feature
 
-#### Proposed Implementation
+#### Implementation
+
+When event show command is called, a pop-up window (an `EventShowWindow` object) will show, displaying the information of the event that corresponds to the given index. The event to show is obtained from the `eventToShowList` instantiated in `ModelManager`.
+
+Given below is an example usage scenario and how the event show mechanism behaves at each step:
+
+Step 1. The user launches the app for the first time, and executes `ecreate n/first event ...` to create a new event. This called method `Model#updateFilteredEventList()` to update both the `filteredEventList` and the `eventToShowList`.
+
+Step 2. The user  executes `ecreate n/second event ...` to create another event. This called the method `Model#updateFilteredEventList()` again to update both the `filteredEventlist` and the `eventToShowList`.
+
+Displayed event list now has two events at index 1 and 2 respectively.
+
+Step 3. The user executes `eshow 1` to read more about the first event created.
+
+The `eshow` command obtains the event at index `1` from the last shown list (`filteredEventList`) as the index is based on the most recent event list displayed to the user before he calls the `eshow` command. This event is the one that will be shown to the user in the pop-up window.
 
-When the app is initialized, the MainApp will first initialize all key components of the app. Specifically in this case:
+The `eshow` command then filters the `eventToShowList` such that it has only one element, which is the event previously obtained.
 
-First, eventToShowList in ModelManager will be initialized, with the list of all events currently in the EventStorage.
+The `eshow` command then returns a `CommandResult` object with its `isShowEvent` field as true.
 
-Then, when MainWindow is being initialized, the EventShowWindow will also be initialized.
+The `MainWindow` received the `CommandResult`, identifies that its `isShowEvent` field is true, hence executing the `MainWindow#handleShowEvent()` method, which shows/focuses the `EventShowWindow`, after loading its contents by calling `EventShowWindow#loadContents()`.
```

Comment:
e.g. here might be too many low-level details, others might wanna chime in too

Reply:
hmm i gets ya but im not sure what to include and what not to as it kinda explains the mechanism behind eshow, any suggestions?

Reply:
This sentence at line 341 has too many commas, shd split it up into a few sentences to make it more understandable for the reader

Reply:
k english teacher

Reply:
@jordankanghm is the diagram i screenshot and put on top sufficient?